src/components/layout/TopBar.jsx

```diff
@@ -34,6 +34,9 @@ class TopBar extends Component {
   isActive = (id) => this.props.activeTab === id;
 
   render() {
+    const { user } = this.props;
+    // Prefer email, then preferred_username, then name, then username
+    const userLabel = (user && (user.email || user.preferred_username || user.name || user.username)) || '';
```

Comment:
We might need an internal sync but imo `preferred_username` should have the highest priorities here among these other fields here. Based on their definations in fence https://github.com/uc-cdis/fence/blob/2a0ede3587ca64cc856991ccdf7c39202e15a2c7/fence/resources/user/__init__.py#L91-L94, I think the order should be `preferred_username > display_name > name > username > email`
And for your use case, instead of trying to patch this from FE side, maybe a better solution is to update the `preferred_username` properly in Fence when the user's record gets created